From 73a19248e71c515c93f331a83e48dcc62896eb03 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Sun, 05 Jun 2022 18:28:01 +0300
Subject: [PATCH] Revert "[SW-77251] common: add memory apis"

Revert rom recovery implementation for secured firmware update via
inband and OOB.

This reverts commit e0474271d358b3bad14a64d2b7bfa0429eb34107.

Change-Id: I6a9d33c0818d34f4aa1eed546c1206b0537ccfa7
---

diff --git a/common/memory.c b/common/memory.c
deleted file mode 100644
index 46a97ec..0000000
--- a/common/memory.c
+++ /dev/null
@@ -1,62 +0,0 @@
-// SPDX-License-Identifier:     GPL-2.0+
-/*
- * Copyright (C) 2022 HabanaLabs Ltd.
- * All Rights Reserved.
- *
- */
-
-#include <arcboot.h>
-#include <memory.h>
-
-static struct mem_desc mem_desc[MEM_NOENT] = {
-	[MEM_GP] = {
-		.base = MEM_GP_BASE,
-		.size = MEM_GP_SZ,
-	},
-	[MEM_LKD_PCI] = {
-		.base = MEM_LKD_PCI_BASE,
-		.size = MEM_LKD_PCI_SZ,
-	},
-	[MEM_FWUPD_PCI] = {
-		.base = MEM_FWUPD_PCI_BASE,
-		.size = MEM_FWUPD_PCI_SZ,
-	},
-	[MEM_I2C] = {
-		.base = MEM_I2C_BASE,
-		.size = MEM_I2C_SZ,
-	},
-};
-
-__weak int arcboot_get_dev_spec_mem_region(__unused enum mem_region region,
-					   __unused struct mem_desc *desc)
-{
-	return 1; /* not dev spec */
-}
-
-int arcboot_get_mem_region(enum mem_region region, struct mem_desc *desc)
-{
-	int err;
-
-	switch (region) {
-	case MEM_GP:
-	case MEM_LKD_PCI:
-	case MEM_FWUPD_PCI:
-	case MEM_I2C:
-		break;
-	default:
-		return 1;
-	}
-
-	if (!desc)
-		return 1;
-
-	/* device defined memory regions */
-	err = arcboot_get_dev_spec_mem_region(region, desc);
-	if (!err)
-		return err;
-
-	desc->base = mem_desc[region].base;
-	desc->size = mem_desc[region].size;
-
-	return 0;
-}
diff --git a/common/memory.h b/common/memory.h
deleted file mode 100644
index 29d28ae..0000000
--- a/common/memory.h
+++ /dev/null
@@ -1,38 +0,0 @@
-/* SPDX-License-Identifier:     GPL-2.0+
- *
- * Copyright (C) 2022 HabanaLabs Ltd.
- */
-
-#ifndef _MEMORY_H_
-#define _MEMORY_H_
-
-#include <stdint.h>
-
-/* TODO: re-verify below offsets */
-#define SDRAM_BASE		0x10000000
-
-#define MEM_GP_BASE		(SDRAM_BASE + 0x40000)	/* SRAM_BASE + 256K */
-#define MEM_GP_SZ		0xC0000			/* 768K */
-#define MEM_LKD_PCI_BASE	(SDRAM_BASE + 0x200000)	/* SRAM_BASE + 2M */
-#define MEM_LKD_PCI_SZ		0x200000		/* 2M */
-#define MEM_FWUPD_PCI_BASE	(SDRAM_BASE + 0x600000)	/* SRAM_BASE + 6M */
-#define MEM_FWUPD_PCI_SZ	0x200000		/* 2M */
-#define MEM_I2C_BASE		(SDRAM_BASE + 0x800000)	/* SRAM_BASE + 8M */
-#define MEM_I2C_SZ		0x200000		/* 2M */
-
-enum mem_region {
-	MEM_GP,
-	MEM_LKD_PCI,
-	MEM_FWUPD_PCI,
-	MEM_I2C,
-	MEM_NOENT	/* terminator */
-};
-
-struct mem_desc {
-	uintptr_t base;
-	uint32_t size;
-};
-
-int arcboot_get_mem_region(enum mem_region region, struct mem_desc *desc);
-
-#endif /* _MEMORY_H_ */
